src/eth_rpc/servers/debug_rpc.rs

```diff
@@ -141,4 +146,38 @@ impl<P: EthereumProvider + Send + Sync + 'static> DebugApiServer for DebugRpc<P>
         // Returns the vector containing the raw receipts
         Ok(raw_receipts)
     }
+
+    /// Returns the Geth debug trace for the given block number.
+    async fn trace_block_by_number(
+        &self,
+        block_number: BlockNumberOrTag,
+        opts: Option<GethDebugTracingOptions>,
+    ) -> Result<Option<Vec<TraceResult>>> {
+        let provider = Arc::new(&self.eth_provider);
+        let maybe_tracer =
+            TracerBuilder::new(provider).await?.with_block_id(BlockId::Number(block_number)).await?.build()?;
+        if maybe_tracer.is_none() {
```

Comment:
why are we doing this here?

why not unwrap_or(Ok(none)) or match?

Reply:
Not sure how you would use unwrap_or and yes we could use a match, just find this more readable

Reply:
```
maybe_tracer
    .map(|tracer| {
        tracer.debug_block(opts.unwrap_or_default())
    })
    .transpose()
```

Reply:
mmmm I find my version more readable no? can change if you want but this won't change anything performance wise